### DISCUSSION TITLE
Error when trying to use Wixext UI on Github build

### DISCUSSION THREAD

**Discussion by na185068** [wixtoolset/issues#7115] [Questions] [answered]
Hi, I am new to WIX, and started my first project (a simple MSI with a "choose install folder" dialog and copy a few files) about two weeks ago.
I am using VS2022 and with the last versions of HeatWave and V4, created a minimalistic project following [Rob Mensching's great video](https://www.youtube.com/watch?v=PXqCZwNPMKE) .
When building on my local computer it works just fine, but when I try to build it on GitHub cl/cd (after pushing the whole project to GItHub) I get the following error:
**D:\b\_work\msi_proj\msi_proj\Folders.wxs(4): error WIX0200: The Fragment element contains an unhandled extension element 'WixUI'. Please ensure that the extension for elements in the 'http://wixtoolset.org/schemas/v4/wxs/ui' namespace has been provided.** 
Can anyone help, since I am struggling with it for already few days ( and nights :))

**Comment by robmen** [comment]
Can you share the repo where you are working? If not, at least the .wixproj? The error indicates that your project does not reference the `WixToolset.UI.wixext` extension, but if it builds locally, it isn't clear why it would fail remotely.

**Reply by na185068** [reply]
Attached is the whole project.

**Comment by na185068** [comment]
[RFUEL_Update.zip](https://github.com/wixtoolset/issues/files/10294038/RFUEL_Update.zip)
Here is the whole project..

**Reply by robmen** [reply]
The `msbuild.binlog` shows a different error:
```
The c:\Users\NA185068\source\repos\Retalix-Fuel-Emerald\.Build\RFUEL_Update\RFUEL_Update\obj\project.assets.json file does not exist. Restore the project before trying to build again. Restore and build with "MSBuild -Restore" or "dotnet build". [c:\Users\NA185068\source\repos\Retalix-Fuel-Emerald\.Build\RFUEL_Update\RFUEL_Update\RFUEL_Update.wixproj]
```

Are you following these instructions in your GitHub action?

> Restore the project before trying to build again. Restore and build with "MSBuild -Restore" or "dotnet build". 

If not, that would explain the other error.

**Reply by na185068** [reply]
Actually, the msbuild.binlog is a leftover of the so many attempts I did on my local machine to simulate the Github build behavior.
I have noticed that the "restore" section in the obj\project.assets.json and in the  obj\ProjectName.wixproj.nuget.dgspec.json files contains my local machine's folders for the Nuget paths including my user name in the path. 
So, what will happen when I call the restore on the Github machine? Won't it fail since it will try to restore using my local paths?  
Anyways, I have uploaded the whole project including the project.assets.json, and got the error I specified at the head of this post..

**Reply by robmen** [reply]
Uhhh, are you committing any files from your `bin` or `obj` folders? If so, don't do that.

**Comment by na185068** [comment, marked as answer]
OK, the solution was: 
1. Rebuild the whole project and change settings of VS: Tools->NuGet Package Manager->Package Manager settings -> Default package management format to "PackageReference" +unchecked All other options besides the "Allow NuGet to download missing packages". 
2. Following Rob's advice, I removed the obj,bin.
3. Run msbuild - restore on the ProjectName.sln
4. Run msbuild the ProjectName.sln.
Hope this will help others who face that issue.
